processes/procstat.go

```diff
@@ -140,12 +142,23 @@ func (psc *procStatsCollector) GetStats(procPath string) (map[string][]Proc, err
 				VmData:  vmData,
 				VmCode:  vmCode,
 			}
-			// tmpName begins and end with brackets, removing them
-			tmpName := strings.Fields(string(procStat))[1]
-			//procName := tmpName[1 : len(tmpName)-1]
-			procName := removeUnwantedChars(tmpName)
-			instances, _ := procs[procName]
-			procs[procName] = append(instances, pc)
+			// procName is process name extracted from command line path
+			procPath := strings.Split(pc.CmdLine, " ")[0]
```

Comment:
It is not possible for returnval to be empty, because `strings.Split` returns single blank element even if provided input, in this case `pc.CmdLine` is empty string:
https://play.golang.org/p/smXbVMAaBF